lib/internal/util/compositekey.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+/**
+ * @param {any} value
+ * @returns {boolean}
+ */
+const hasLifetime = (value) => {
+  return value !== null && (
+    typeof value === 'object' ||
+    typeof value === 'function'
+  );
+}
+class CompositeNode {
+  /**
+   * @type {Map<any, Map<number, CompositeNode>>}
+   */
+  primitiveNodes;
+  /**
+   * @type {null | Readonly<{}>}
+   */
+  value;
+  constructor() {
+    this.value = null;
+  }
+  get() {
+    if (this.value === null) {
+      return this.value = Object.freeze({__proto__: null});
+    }
+    return this.value;
+  }
+  /**
+   * 
+   * @param {any} value 
+   * @param {number} position 
+   */
+  emplacePrimitive(value, position) {
+    if (!this.primitiveNodes) {
+      this.primitiveNodes = new Map();
+    }
+    if (!this.primitiveNodes.has(value)) {
+      this.primitiveNodes.set(value, new Map());
```

Comment:
```suggestion
      this.primitiveNodes = new SafeMap();
    }
    if (!this.primitiveNodes.has(value)) {
      this.primitiveNodes.set(value, new SafeMap());
```

Reply:
SafeMap is aliased as Map at top of file

Reply:
ah, missed that. the alias seems like it would cause this very same confusion tho :-) using SafeMap inline would convey/reassure proper intrinsics usage.